bids/modeling/statsmodels.py

```diff
@@ -698,6 +698,9 @@ def _build_contrasts(self, unique_in_contrast=None):
                 conditions = col_names
 
             for col_name in conditions:
+                if col_name == "intercept":
```

Comment:
I'm worried that this might end up over-correcting. I think we basically need a unit test that shows that:

```YAML
{
  ...,
  "Model": {
    "X": ["a", 1]
  },
  "Contrasts": [
    {"Name": "a", "ConditionList": ["a"], "Weights": [1], "Test": "t"},
    {"Name": 1, "ConditionList": [1], "Weights": [1], "Test": "t"}
  ]
}
```

Produces identical `ContrastInfo`s as each of

```YAML
{
  ...,
  "Model": {
    "X": ["a", 1]
  },
  "DummyContrasts": {"Test": "t"}
}
```

```YAML
{
  ...,
  "Model": {
    "X": ["a", 1]
  },
  "DummyContrasts": {"Contrasts": ["a", 1], "Test": "t"}
}
```

At both the run and subject levels.

Reply:
Oh, and also if there is an explicit `"intercept"` variable, instead of a `1`.

Reply:
Trying to dig this bag up.

Test makes sense. 

But I'm confused: what should be the behavior if there is an explicit "intercept" variable? That is not a "special" key (that's what `1` is for). 

I suppose what will happen is if its in `DummyContrasts` it will be converted to `1` and then back to the intercept name (which should be `intercept`). I think that'll be fine, but it does feel funny because the name `intercept` is not reserved, but it's being treated that way.

Maybe the term `intercept` _should_ be reserved, since it necessarily refers to the intercept in the design matrix (even before my changes it did at higher levels).

Reply:
That said, I think your issue persists-- but it was an issue before also. That is, if there is a variable in X named "intercept" that is *not* the automatically created "intercept" column, it could clash.

That is if your X looks like: `[1, "intercept"]`, you could not actually model both right now. 

I'm fairly sure this was an issue before my changes, since the only thing I changed was that the column named "intercept" would not be renamed to the contrast name. but in the first level this would have still been an issue. 

I can add variable named "intercept" to an event files in the tests to see what happens, but IMO that's a separate issue

Reply:
Okay. If it's not handleable, then that's fine. I think we used `1` to avoid clashing with `"intercept"`, but it is a dumb idea to manually use your own `"intercept"` column that could not be safely overridden by a column of `1`s.

Reply:
I think what we should do is throw an error at the DM creation if you include both `1` and "intercept" in your `X`. If we get past that, I think the rest should not cause any issues.